theme-utils.mjs

```diff
@@ -253,6 +260,53 @@ async function escapePatterns() {
 	}
 }
 
+async function validateSchema(files) {
+	const ajv = new Ajv({
+		allowMatchingProperties: true,
+		allErrors: true,
+	});
+	const errors = [];
+	let progress = progressbar.filledBar(files.length, 0)[0];
+	process.stdout.write(`${progress} 0/${files.length}`, 'utf-8');
+	for (let i = 0; i < files.length; i++) {
+		const filename = files[i];
+		try {
+			const file = await fs.promises
+				.readFile(filename, 'utf-8')
+				.then(JSON.parse);
+			const schemaURL = file.$schema;
+			let schema;
+			if (!schemaURL) {
+				schema = {
+					type: 'object',
+					required: ['$schema'],
+				};
+			} else if (schemaURL.startsWith('http')) {
+				schema = await fetch(schemaURL).then((res) => res.json());
+			} else {
+				schema = await fs.promises
+					.readFile(
+						path.join(path.dirname(filename), schemaURL),
+						'utf-8'
+					)
+					.then(JSON.parse);
+			}
+			if (!ajv.validate(schema, file)) {
+				throw ajv.errors;
+			}
+		} catch (error) {
+			errors.push({ file: filename, error });
+		}
+		progress = progressbar.filledBar(files.length, i + 1)[0];
+		process.stdout.write(`\r${progress} ${i + 1}/${files.length}`, 'utf-8');
+	}
+	console.log();
```

Comment:
should we remove this one?

Reply:
This is the newline after the final `process.stdout.write` from the progressbar which doesn't include one.